### ISSUE TITLE
More challenges

### ISSUE THREAD

**Issue by interrogator** [interrogator/wwc#13]
The students responded very well to an exercise/challenge today, where they searched a tagged corpus for complicated things, like 'pronouns occurring before modals', or 'tokens used as both a noun and verb'.

These kinds of challenges, though, can potentially alienate people who are struggling to keep up. Today's went well because:

* We created a markdown cell and clearly wrote out the challenge parameters
* There were multiple alternatives, ranging in difficulty from easy to very hard, but based on a similar kind of logic
* The results were interesting
* It was before a break, so people could go and grab a coffee once they'd had enough, or work on it for a few extra minutes
* Students could write their solutions on the instructor's machine for all to see

A challenge before the break in every session seems like a good idea, so long as it adheres to the above principles. We need to revise earlier sessions to fit with this.

**Comment by chozelinek** [comment]
I agree. Challenges force students to think computationally. I think that is crucial to have helpers around when we try that. A soft variant of the challenge can be to provide a template with the logic to be filled in. And an advanced one, they have to think the logic themselves.

**Comment by alvations** [comment]
More challenges is good. Helpers is a good solution to challenges, most coding tutorials have some helps around too =)
